components/spocos.js

```diff
@@ -0,0 +1,330 @@
+import styled from 'styled-components';
+import React, { useState, useEffect } from 'react';
+import Card, {
+  CardHeader,
+  CardTitle,
+  CardContent,
+  CardButtonContainer,
+  CardDiv,
+  TableDiv,
+} from './card';
+import Button from './button';
+import { EDIT, NEW, VIEW, DELETE, COLOR, SPONSORSHIP } from '../constants';
+import {
+  getSponsors,
+  deleteSponsorImagefromStorage,
+  deleteSponsor,
+  updateSponsor,
+  uploadSponsorImageToStorage,
+} from '../utility/firebase';
+import { useAuth } from '../utility/auth';
+import Modal, {
+  Label,
+  ModalContent,
+  ModalField,
+  LogoImage,
+  UploadContainer,
+} from './modal';
+
+// TODO: move font family higher, reduce need to redefine
+
+const Text = styled.p`
+  padding-right: 80px;
+  flex: 1;
+  font-size: 16px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+const Actions = styled.div`
+  padding-right: 80px;
+  flex: 1;
+  display: flex;
+  justify-content: flex-start;
+`;
+
+export default function SponsorshipPage(props) {
```

Comment:
no big deal but you could change this so you don't need to type `props.name` you can use `name` in the file
```suggestion
export default function SponsorshipPage({name}) {
```